### PR TITLE
feat(A2-234): lpa statement journey set up and statement question

### PR DESCRIPTION
**Pull request by DeanH1990**
## Ticket Number

https://pins-ds.atlassian.net/browse/AAPD-234

## Description of change

sets up new journey for lpa statement submission and adds appeal statement question

## Checklist

- [ ] Requires infrastructure changes
- [ ] I have updated the documentation accordingly
- [x] My commit history in this PR is linear
- [x] New features have tests
- [ ] Breaking change (team conversation required)

## Important

Please do not merge from `main` (please only [rebase](https://github.com/Planning-Inspectorate/appeal-planning-decision/wiki/An-intro-to-Git-Rebase)). This keeps the history linear and easier to debug.

### CODE REVIEW

File: packages/appeals-service-api/src/routes/v2/appeal-cases/_caseReference/lpa-statement-submission/index.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+const express = require('express');
+const {
+	getLPAStatementSubmission,
+	patchLPAStatementSubmission,
+	createLPAStatementSubmission
+} = require('./controller');
+const asyncHandler = require('@pins/common/src/middleware/async-handler');
+const router = express.Router({ mergeParams: true });
+
```
Comment (bryn-pins) [inline packages/appeals-service-api/src/routes/v2/appeal-cases/_caseReference/lpa-statement-submission/index.js:9]: can we add auth to the endpoints please
generic token auth
and ownership check checking user belongs to lpa the case is assigned to

submissionRepo.lpaCanModifyCase

**Review by bryn-pins — COMMENTED** [review]
can we get some api tests please

File: packages/forms-web-app/src/dynamic-forms/middleware/redirect-to-unanswered-question.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+	const journeyResponse = res.locals.journeyResponse;
+	const journey = getJourney(journeyResponse);
+
+	for (const section of journey.sections) {
+		for (const question of section.questions) {
+			const answer = journey.response?.answers[question.fieldName];
+			if (!answer || answer === '' || answer === undefined) {
+				let redirectUrl;
+
+				if (section.segment === config.dynamicForms.DEFAULT_SECTION) {
```
Comment (bryn-pins) [inline packages/forms-web-app/src/dynamic-forms/middleware/redirect-to-unanswered-question.js:14]: did having a single section with no url not work then?

Comment (DeanH1990) [reply packages/forms-web-app/src/dynamic-forms/middleware/redirect-to-unanswered-question.js:14]: I couldnt get it to work, not without having to make changes throughout the journey/ question classes, the controllers and validators as the section is pretty embedded into how everything works. 

This is the route i was originally going down until we had the discussion last week. I wasnt removing the coupling though, just adding an if check more or less anywhere the section/segment was used and putting null as the argument for the section. It definitely wasnt optimal.

This currently uses a default section which allows all the existing functions to work as they are. The section is just not used in the url as there is only the one.

File: packages/forms-web-app/src/dynamic-forms/journey.js [hunk 3]
```diff
@@ -250,6 +250,15 @@ class Journey {
 		);
 	};
 
+	/**
+	 * Gets the url for the current question
+	 * @param {string} questionSegment - question segment name
+	 * @returns {string} url for the current question
+	 */
+	getCurrentQuestionUrlWithoutSection = (questionSegment) => {
+		return `${this.baseUrl}/${encodeURIComponent(questionSegment)}`;
```
Comment (bryn-pins) [inline packages/forms-web-app/src/dynamic-forms/journey.js:259]: bit concerned we're not centralizing the logic in buildQuestionUrl, anything that relies on that function will fail right?

Comment (DeanH1990) [reply packages/forms-web-app/src/dynamic-forms/journey.js:259]: Ah sorry, must've had a brain fart. Did this as the 'Change' / 'Answer' button on the task list/ check your answers page was adding the section back in to the url. Was thinking the getCurrentQuestionUrl wouldnt make it to the buildQuestionUrl due to the check of no section... 

Will change it

**Review by bryn-pins — APPROVED** [review]
might be some improvements we can make regarding the urls/no sections
but overall looks good

**Pull request merged**